tests/Algorithms/HungarianAlgorithmTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using Bearded.Utilities.Algorithms;
+using FsCheck.Xunit;
+using Xunit;
+
+namespace Bearded.Utilities.Tests.Algorithms
+{
+    public class HungarianAlgorithmTests
+    {
+        [Fact]
+        public void Run_EmptyCostMatrix_DoesntFail()
+        {
+            Assert.Empty(HungarianAlgorithm.Run(new float[0, 0]));
+        }
+
+        [Property]
+        public void Run_OneSource_AssignsTarget(float cost)
+        {
+            if (float.IsInfinity(cost) || float.IsNaN(cost))
+                Assert.Throws<ArgumentException>(() => HungarianAlgorithm.Run(new[,] {{cost}}));
+            else
+                Assert.Equal(new[] { 0 }, HungarianAlgorithm.Run(new[,] { { cost } }));
+        }
+
+        [Fact]
+        public void Run_TwoSources_AssignsLowestCostTarget()
```

Comment:
This should be a property test I think. Otherwise it does not really test what it claims.